.circleci/config.yml

```diff
@@ -31,6 +31,25 @@ references:
         source ~/infrastructure/venv/ansible/bin/activate
         echo "source ~/infrastructure/venv/ansible/bin/activate" >> $BASH_ENV
 
+  setup_infra: &setup_infra
+    run:
+      name: Setup infrastructure
+      command: |
+        cd ansible && ansible-playbook -e 'ansible_python_interpreter="/usr/bin/env python"' \
+          --tags "${DEPLOY_ENV:?}" environments.yml
+
+  setup_node: &setup_node
+    run:
+      name: Setup environment
+      command: |
+        cd ansible && ansible-playbook --limit="tag_env_${DEPLOY_ENV:?}:&tag_role_epoch" setup.yml
```

Comment:
I see [ `:&` in host pattern is intersection (while simple `:` in host pattern is OR)](https://docs.ansible.com/ansible/2.4/intro_patterns.html).

I see labels tag_role_epoch and tag_env* (e.g. tag_env_uat) are assigned to hosts in https://github.com/aeternity/infrastructure/blob/56cfd6326b59f8e91afe53397b7c1d2f5ae11eae/ansible/environments.yml#L55 though it is not clear to me how meta_group works.

What concerns me is that I do not see these labels assigned to AWS nodes https://github.com/aeternity/infrastructure/blob/56cfd6326b59f8e91afe53397b7c1d2f5ae11eae/ansible/environments.yml#L117-L169 (and I do not see dump_inventory called anywhere by CI) hence AWS nodes excluded. Correct? How does this process extend to AWS nodes from CI then?

Reply:
We setup the same AWS tags in https://github.com/aeternity/infrastructure/blob/master/aws/ae-environment.yml#L99-L117